ttnn/cpp/ttnn/operations/experimental/transformer/nlp_create_qkv_heads/device/nlp_create_qkv_heads_device_operation.cpp

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "nlp_create_qkv_heads_device_operation.hpp"
+
+#include "ttnn/deprecated/tt_dnn/op_library/work_split.hpp"
+
+namespace ttnn::operations::experimental::transformer {
+
+// Generic NLP CreateHeads op
+void NlpCreateHeadsDeviceOperation::validate_on_program_cache_miss(const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
```

Comment:
Can we add more explanation where we can.
Fir example instead of "Unsupported input shape" we can provide exact shape  and show that it is < then TILE_HEIGHT

Reply:
Done